types/marked/index.d.ts

```diff
@@ -17,7 +18,7 @@ export = marked;
  * @param callback Function called when the markdownString has been fully parsed when using async highlighting
  * @return String of compiled HTML
  */
-declare function marked(src: string, callback: (error: any | undefined, parseResult: string) => void): string;
+declare function marked(src: string, callback?: (error: any | undefined, parseResult: string) => void): string | void;
```

Comment:
It seems like you get a string if there's no callback and void if there is.  If that's the case, could you represent it using a pair of overloads with those shapes?

Reply:
Good call! Implemented this for the `marked` default export and added a more detailed description to `marked.MarkedOptions.highlight`

Reply:
I may just be missing something, but why not make the same change for `highlight`?

Reply:
I tried, I couldn't get it to work (see https://stackoverflow.com/questions/60519455/typescript-function-overloading-in-interface).

Here's the exact error I was seeing:

```
marked-tests.ts:19:1 - error TS2322: Type '(code: string, lang: string, callback: (error: any, code?: string) => void) => void' is not assignable to type '{ (code: string, lang: string): string; (code: string, lang: string, callback: (error: any, code?: string) => void): void; }'.
```

Reply:
Huh, you learn something new every day.  So, it turns out the type declaration you'd use for assignment doesn't work for calling and vice versa.  I'd just leave it as is.  Thanks for the effort though!